styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java

```diff
@@ -82,13 +82,16 @@ private static int getExitCode(Pod pod, ContainerStatus status) {
         return message.exitCode;
       } catch (IOException e) {
         LOG.warn("Unexpected termination log message for container {}", status.getContainerID(), e);
-
-        // make sure to signal an error to be on the safe side, as opposed to the purported exit code
-        return 127;
+        return NO_EXIT_CODE;
       }
     }
 
-    return terminated.getExitCode();
+    if (terminated.getExitCode() == null) {
```

Comment:
Started thinking about whether we need this branch. We are not checking existence of `status.getState().getTerminated()` or so. Maybe we should trust kube.